src/flexible_type/flexible_type_converter.hpp

```diff
@@ -10,657 +10,899 @@
+  template <typename T> static constexpr bool matches() { return false; }
+
+  template <typename T> static inline void get(T& t, const flexible_type& v) {
+    static_assert(swallow_to_false<T>::value, "get with bad match.");
+  }
+
+  template <typename T>
+  static inline void set(flexible_type& t, const T& v) {
+    static_assert(swallow_to_false<T>::value, "set with bad match.");
+  }
 };
 
-/**
- * Case 1a: Cover all the direct members of flexible_type, 
- *         EXCEPT for numeric types (flex_int, flex_float), and flex_list.
- *        - flex_string (std::string)
- *        - flex_vec (std::vector<double>)
- *        - flex_dict (std::vector<std::pair<flexible_type, flexible_type> >)
- *        - flex_date_time (boost::ptime)
+////////////////////////////////////////////////////////////////////////////////
+//
+//  flexible_type
+//
+////////////////////////////////////////////////////////////////////////////////
+
+/** straight flexible_type.  Always given priority.
  */
-template <typename T>
-struct flexible_type_converter<T,
-    typename std::enable_if<is_flexible_type_member<T>::value &&
-                            !std::is_arithmetic<T>::value &&
-                            !std::is_same<T, flex_list>::value>::type> {
-  static constexpr bool value = true;
-  T get(const flexible_type& val) {
-    if (type_to_enum<T>::value != val.get_type()) {
-      std::string errormsg = std::string("Expecting ")
-          + flex_type_enum_to_name(type_to_enum<T>::value) 
-          + ". But we got a " + flex_type_enum_to_name(val.get_type());
-      throw(errormsg);
-    }
-    return val.get<T>();
+template <> struct ft_converter<CVTR__FLEXIBLE_TYPE> {
+
+  template <typename T> static constexpr bool matches() {
+    return std::is_same<T, flexible_type>::value;
+  }
+
+  static void get(flexible_type& dest, const flexible_type& src) {
+    dest = src;
   }
-  flexible_type set(const T& val) {
-    return flexible_type(val);
+
+  static void set(flexible_type& dest, const flexible_type& src) {
+    dest = src;
   }
 };
 
+////////////////////////////////////////////////////////////////////////////////
+//
+//  numeric types.
+//
+////////////////////////////////////////////////////////////////////////////////
 
-/**
- * Case 1b: flex_list: std::vector<flexible_type> 
+/** Any floating point values.
  */
-template <>
-struct flexible_type_converter<flex_list, void> {
-  static constexpr bool value = true;
-  flex_list get(const flexible_type& val) {
-    if (val.get_type() == flex_type_enum::LIST) {
-      return val;
-    }
-    else if (val.get_type() == flex_type_enum::VECTOR) {
-      flex_vec f = val.to<flex_vec>();
-      flex_list ret;
-      ret.resize(f.size());
-      for (size_t i = 0;i < f.size(); ++i) ret[i] = f[i];
-      return ret;
+template <> struct ft_converter<CVTR__FLOATING_POINT> {
+
+  template <typename T> static constexpr bool matches() {
+    return std::is_floating_point<T>::value;
```

Comment:
Why is this and Integer case separate? Why not merge as an std::is_arithmetic test?